main.go

```diff
@@ -12,13 +12,21 @@ import (
 
 func makeHandler(name string, fn func(http.ResponseWriter, *http.Request, string, string)) http.HandlerFunc {
```

Comment:
Line 30 `fn(w, r, m[1], m[2])` seems to expect `m[1]` and `m[2]` to be available slice indices.  Now if thats a good thing or not is a different question. 

I assumed that line 17 `validPath.FindStringSubmatch(r.URL.Path)` returns `nil` if `validPath` can't produce a slice of `m[0, 1, 2]` so that is why line 30 can expect `m[2]` to be valid. 

I say all this assuming the same things you mentioned above :sweat_smile: